infra/base-images/base-builder/jcc/jcc.go

```diff
@@ -166,9 +166,31 @@ func CorrectMissingHeaders(bin string, cmd []string) (bool, error) {
 	return false, nil
 }
 
-func GenerateAST(bin string, args []string, targetFile string) {
+func EnsureDir(dirPath string) bool {
+	// Checks if a path is an existing directory, otherwise create one.
+	pathInfo, err := os.Stat(dirPath)
+	if err == nil {
+		isDir := pathInfo.IsDir()
+		if !isDir {
+			fmt.Println(dirPath, "exists but is not a directory.")
+		}
+		return isDir
+	}
+
+	if os.IsNotExist(err) {
```

Comment:
per https://pkg.go.dev/os#IsNotExist, can you use ` errors.Is(err, fs.ErrNotExist)` instead?